cmd/kueuectl/app/list/list_workload.go

```diff
@@ -0,0 +1,311 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package list
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/genericiooptions"
+	"k8s.io/cli-runtime/pkg/printers"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"sigs.k8s.io/kueue/apis/kueue/v1beta1"
+	"sigs.k8s.io/kueue/apis/visibility/v1alpha1"
+	clientset "sigs.k8s.io/kueue/client-go/clientset/versioned"
+	"sigs.k8s.io/kueue/client-go/clientset/versioned/scheme"
+	"sigs.k8s.io/kueue/cmd/kueuectl/app/util"
+	"sigs.k8s.io/kueue/pkg/workload"
+)
+
+const (
+	wlLong    = `Lists Workloads that match the provided criteria.`
+	wlExample = `  # List Workload 
+  kueuectl list workload`
+)
+
+type WorkloadOptions struct {
+	PrintFlags *genericclioptions.PrintFlags
+
+	OnlyPending        bool
+	OnlyAdmitted       bool
+	AllNamespaces      bool
+	Namespace          string
+	FieldSelector      string
+	LabelSelector      string
+	ClusterQueueFilter string
+	LocalQueueFilter   string
+
+	ClientSet clientset.Interface
+
+	ToPrinter func(r *listWorkloadResources) (printers.ResourcePrinterFunc, error)
+
+	genericiooptions.IOStreams
+}
+
+func NewWorkloadOptions(streams genericiooptions.IOStreams) *WorkloadOptions {
+	return &WorkloadOptions{
+		PrintFlags: genericclioptions.NewPrintFlags("").WithTypeSetter(scheme.Scheme),
+		IOStreams:  streams,
+	}
+}
+
+func NewWorkloadCmd(clientGetter util.ClientGetter, streams genericiooptions.IOStreams) *cobra.Command {
+	o := NewWorkloadOptions(streams)
+
+	cmd := &cobra.Command{
+		Use: "workload [-–clusterqueue CLUSTER_QUEUE_NAME] [-–localqueue LOCAL_QUEUE_NAME] [--only-pending] [--only-admitted] [--all-namespaces]",
```

Comment:
I think we're missing to indicate the fieldselector and labelselector flags.

Reply:
Thanks. Added.